### DISCUSSION TITLE
IIS setting?

### DISCUSSION THREAD

**Discussion by amir-inbar-software** [Mimetis/Dotmim.Sync#531] [Q&A]
Hi, i set up an ASP.NET Web Core API project and managed to sync data.
When i publish the project to to IIS (not the express IIS) i  can't sync the data anymore and the sync operation hangs...

When i try to see the controller data in the address "....mywebsite/api/sync" i get the following message:

"Server is configured to Production mode. No options displayed"

am i doing something wrong? how can i debug the project?

Reagrds.

**Comment by Mimetis** [comment]
Compile in `Debug` mode instead of `Release` and deploy to your ASP.Net Core host.
You will have access to the sync summary web page.

Othewise, check that your SQL Server instance is reachable from your ASP.Net Core host

**Reply by amir-inbar-software** [reply]
Thanks a lot for the reply.

When i deploy the debug configuration to the IIS i still get the "Server is configured to Production mode. No options displayed" in the controller address... and don't have the access to summary webpage.

I have waited long enough when i tried another sync and got this message - "Dotmim.Sync.Web.Client.HttpSyncWebException: 'Login failed for user 'IIS APPPOOL\AirCo365WebSync'.'"

Do have a clue?

**Reply by Mimetis** [reply]
I guess you don't have access to your database server from your asp.net core website

If you are using `Integrated Security=true` to connect to your database, it's normal that you have no access with the IIS User.
If it's the reason why you can't connect, you may consider using a sql server user (`Integrated Security=false; User Id=XXXX; Password=XXX`)

One test you can do is to deploy a simple asp.net core web api that tries to connect to your web server and see if you have any error raised ?

A part from that, if you still want to have the sync web page, you need to configure the `ASPNETCORE_ENVIRONMENT` environment variable to `Development` instead of `Production` (Do it only to debug your solution, then go back to `Production`)

**Reply by amir-inbar-software** [reply]
Yes, i tried a custom user the sql server DB:

"ConnectionStrings": {
    "DefaultConnection": "Data Source=AMIR-SP7\\SP7;Initial Catalog=dmsSync;Integrated Security=false;User Id=dmsTest;Password=123456;"
  }

but still same message - "Login failed for user 'IIS APPPOOL\AirCo365WebSync'.'"

So i did something else....   i added to the SQL server another user name - "IIS APPPOOL\AirCo365WebSync" with permissions to dmsSync DB and now it's working! :)

looks like that under IIS the dotmim.sync.web.client component tries to connect with that user "IIS APPPOOL/" + project name (AirCo365WebSync is my API project name)

Is that make any sense to you?

**Reply by amir-inbar-software** [reply]
another update like you said, but this is how it worked: in order to see the summary web page in the IIS after deploy i had to add to the web.config the following:

< environmentVariables>
                < environmentVariable name="ASPNETCORE_ENVIRONMENT" value="Development" />
< /environmentVariables>

Regards.
Amir

**Reply by Mimetis** [reply]
> looks like that under IIS the dotmim.sync.web.client component tries to connect with that user "IIS APPPOOL/" + project name (AirCo365WebSync is my API project name)
> Is that make any sense to you?

Nop; `DMS` will only uses the connection string you are providing, nothing else :)
